plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2083,15 +2083,16 @@ void block_only_sync::schedule() {
       // wait one second to see if we can actually get the block from net plugin before we try to resync from block vault
       _start_sync_timer.expires_from_now(boost::posix_time::seconds(1));
       _pending = true;
-      _start_sync_timer.async_wait(
-          app().get_priority_queue().wrap(priority::high, [this](const boost::system::error_code& ec) {
-             if (!ec) {
+      _start_sync_timer.async_wait(app().get_priority_queue().wrap(
+          priority::high, [this, weak_impl = _impl->weak_from_this()](const boost::system::error_code& ec) {
+             auto shared_impl = weak_impl.lock();
+             if (shared_impl.get() && !ec) {
```

Comment:
While I understand that `weak_impl.lock()` should protect situations were `this` is stale it seems sloppy to use `this->_impl` rather than `shared_impl`.  I would even go so far as to route `this->_pending = false` through that `shared_ptr<>` as well.